<launch>
  <arg name="map_file" />

  <!--- Run mapserver -->
  <include file="$(find alufr_navigation)/launch/map_server.launch">
    <arg name="map_file" value="$(arg map_file)" />
  </include>
  
  <!--- Run AMCL -->
  <include file="$(find amcl)/examples/amcl_omni.launch" />

  <!--- Run move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find alufr_navigation)/config/move_base/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find alufr_navigation)/config/move_base/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find alufr_navigation)/config/move_base/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find alufr_navigation)/config/move_base/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find alufr_navigation)/config/move_base/base_local_planner_trajectory_ros.yaml" command="load" />
    <param name="controller_frequency" value="15.0" />
    <!--  Remapping needed in case the safety node below is used  <remap from="cmd_vel" to="robotino_cmd_vel" />-->
  </node>
  
  <!-- Run rviz -->
  <include file="$(find squirrel_navigation)/launch/rviz.launch">
    <arg name="mapping" value="false"/>
  </include>
</launch>
